packages/react-devtools-shared/src/__tests__/store-test.js

```diff
@@ -1008,4 +1010,197 @@ describe('Store', () => {
       done();
     });
   });
+
+  describe('inline errors and warnings', () => {
+    test('during render are counted', () => {
+      function Example() {
+        console.error('test-only: render error');
+        console.warn('test-only: render warning');
+        return null;
+      }
+      const container = document.createElement('div');
+
+      withErrorsOrWarningsIgnored(['test-only:'], () => {
+        act(() => ReactDOM.render(<Example />, container));
+      });
+
+      expect(store.errorsAndWarnings).toEqual(
+        new Map([[2, {errors: 1, warnings: 1}]]),
+      );
+
+      withErrorsOrWarningsIgnored(['test-only:'], () => {
+        act(() => ReactDOM.render(<Example rerender={1} />, container));
+      });
+
+      expect(store.errorsAndWarnings).toEqual(
+        new Map([[2, {errors: 2, warnings: 2}]]),
+      );
+    });
+
+    test('during layout get counted', () => {
+      function Example() {
+        React.useLayoutEffect(() => {
+          console.error('test-only: layout error');
+          console.warn('test-only: layout warning');
+        });
+        return null;
+      }
+      const container = document.createElement('div');
+
+      withErrorsOrWarningsIgnored(['test-only:'], () => {
+        act(() => ReactDOM.render(<Example />, container));
+      });
+
+      expect(store.errorsAndWarnings).toEqual(
+        new Map([[2, {errors: 1, warnings: 1}]]),
+      );
+
+      withErrorsOrWarningsIgnored(['test-only:'], () => {
+        act(() => ReactDOM.render(<Example rerender={1} />, container));
+      });
+
+      expect(store.errorsAndWarnings).toEqual(
+        new Map([[2, {errors: 2, warnings: 2}]]),
+      );
+    });
+
+    test('during passive get counted', () => {
+      function Example() {
+        React.useEffect(() => {
+          console.error('test-only: passive error');
+          console.warn('test-only: passive warning');
+        });
+        return null;
+      }
+      const container = document.createElement('div');
+
+      withErrorsOrWarningsIgnored(['test-only:'], () => {
+        act(() => ReactDOM.render(<Example />, container));
+      });
+
+      expect(store.errorsAndWarnings).toEqual(
+        new Map([[2, {errors: 1, warnings: 1}]]),
+      );
+
+      withErrorsOrWarningsIgnored(['test-only:'], () => {
+        act(() => ReactDOM.render(<Example rerender={1} />, container));
+      });
+
+      expect(store.errorsAndWarnings).toEqual(
+        new Map([[2, {errors: 2, warnings: 2}]]),
+      );
+    });
+
+    // TODO (inline errors) The fiber for the recorded error is considered to be not mounted by renderer.js#isFiberMounted.
+    test('from react get counted', () => {
+      const container = document.createElement('div');
+      function Example() {
+        return <div data-camelCased="should be lowercase" />;
+      }
+
+      withErrorsOrWarningsIgnored(
```

Comment:
`inspectedElementContext-test` does not need this currently because there are tests that mock console methods without restoring the original method e.g. https://github.com/facebook/react/blob/9380545c50e3b14a0e3a898c1767beb17f11f652/packages%2Freact-devtools-shared%2Fsrc%2F__tests__%2FinspectedElementContext-test.js#L389.

IMO these tests should make sure they restore their mocks properly. If you only run the newly added tests in `inspectedElementContext-test` you'll notice how you get the calls to console.error logged which I would consider noise.

The new test uitls looks a bit hacky though. I just wanted to make sure we don't accidentally silence unexpected calls to console error or message.

Reply:
Good call. If you see tests that don't tear down mocks correctly, please feel free to fix them (or at least add a TODO and tag me on it)

Reply:
Have it locally. Needed a console.log that was mocked in an earlier test :grin: